src/Dfe.PlanTech.Application/Persistence/Commands/WebhookToDbCommand.cs

```diff
@@ -0,0 +1,178 @@
+                logger.LogInformation("Processing minimal payload event {CmsEvent} for entity with ID {Id}", mapped.CmsEvent, mapped.IncomingEntity.Id);
+                await ProcessEntityRemovalEvent(mapped, cancellationToken);
+            }
+            else if (!mapped.IsValid)
+            {
+                logger.LogWarning("Entity {MappedEntityType} is invalid", mapped.IncomingEntity?.GetType());
+            }
+            else
+            {
+                UpsertEntity(mapped);
+                await DbSaveChanges(cancellationToken);
+            }
+
+            cacheClearer.ClearCache();
+            return new ServiceBusSuccessResult();
+        }
+        catch (Exception ex) when (ex is JsonException or CmsEventException)
+        {
+            logger.LogError(ex, "Error processing message ID {Message}", id);
+            return new ServiceBusErrorResult(ex.Message, ex.StackTrace, false);
+        }
+        catch (Exception ex)
+        {
+            return new ServiceBusErrorResult(ex.Message, ex.StackTrace, true);
+        }
+    }
+
+    protected virtual Task<int> ProcessEntityRemovalEvent(MappedEntity mapped, CancellationToken cancellationToken)
+    {
+        if (mapped.ExistingEntity == null)
+        {
+            throw new InvalidOperationException("ExistingEntity is null for removal event but various validations should have prevented this.");
+        }
+
+        return databaseHelper.Database.SetComponentPublishedAndDeletedStatuses(mapped.ExistingEntity, mapped.IncomingEntity.Published, mapped.IncomingEntity.Deleted, cancellationToken);
+    }
+
+    /// <summary>
+    /// Checks if the message with the given CmsEvent should be ignored based on certain conditions.
+    /// </summary>
+    /// <remarks>
+    /// If we receive a create event, we return true.
+    /// If we are NOT using preview mode (i.e. we are ignoring drafts), and the event is just a save or autosave, then return true.
+    ///
+    /// Else, we return false.
+    /// </remarks>
+    /// <param name="cmsEvent"></param>
+    /// <param name="isPublished"></param>
+    /// <returns></returns>
+    private bool ShouldIgnoreMessage(CmsEvent cmsEvent, bool isPublished)
+    {
+        switch (cmsEvent)
+        {
+            case CmsEvent.CREATE:
+                logger.LogInformation("Dropping received event {CmsEvent}", cmsEvent);
+                return true;
+            case CmsEvent.SAVE or CmsEvent.AUTO_SAVE when isPublished && !contentfulOptions.UsePreview:
+                logger.LogInformation("Received {Event} but UsePreview is {UsePreview} - dropping message", cmsEvent,
+                    contentfulOptions.UsePreview);
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the CmsEvent based on the provided subject.
+    /// </summary>
+    /// <param name="contentfulEvent">The Contentful event (from the service bus message subject)</param>
```

Comment:
Trivial suggestion, but I reckon it wouldn't hurt to add `e.g. ContentManagement.Entry.publish` in param or summary to help a reader understand the parsing